src/main/java/com/google/cloud/tools/gradle/appengine/AppEngineStandardPlugin.java

```diff
@@ -67,29 +70,24 @@
   public void apply(Project project) {
     project.getPluginManager().apply(WarPlugin.class);
 
-    // We can't configure war tasks in model space because the War plugin is not a RuleSource plugin
-    // so do it here for now till Gradle updates all the core plugins.
-    createExplodedWarTask(project);
+    // Create an extension to share data from project space to model space
+    final StandardDataExtension projectData = project.getExtensions()
+        .create("_internalProjectData", StandardDataExtension.class);
+
+    configureTargetCompatibility(project, projectData);
   }
 
-  private void createExplodedWarTask(final Project project) {
-    final War warTask = (War) project.getTasks().getByName(WarPlugin.WAR_TASK_NAME);
-    project.getTasks()
-        .create(EXPLODE_WAR_TASK_NAME, ExplodeWarTask.class, new Action<ExplodeWarTask>() {
-          @Override
-          public void execute(final ExplodeWarTask explodeWarTask) {
-            explodeWarTask
-                .setExplodedAppDirectory(new File(project.getBuildDir(), EXPLODED_APP_DIR_NAME));
-            explodeWarTask.dependsOn(warTask);
-            project.getTasks().getByName(BasePlugin.ASSEMBLE_TASK_NAME).dependsOn(explodeWarTask);
-            project.afterEvaluate(new Action<Project>() {
-              @Override
-              public void execute(Project project) {
-                explodeWarTask.setWarFile(warTask.getArchivePath());
-              }
-            });
-          }
-        });
+  private void configureTargetCompatibility(final Project project,
+      final StandardDataExtension projectData) {
+    project.afterEvaluate(new Action<Project>(){
+      @Override
+      public void execute(Project project){
+        JavaPluginConvention javaConvention = project.getConvention()
+            .getPlugin(JavaPluginConvention.class);
+        JavaVersion javaVersion = javaConvention.getTargetCompatibility();
```

Comment:
Can getTargetCompatibility() return null, if targetCompatibility is not specified in build.gradle?